src/main/typescript/actor/wvActor.ts

```diff
@@ -476,58 +482,82 @@ export default class WvActor extends Actor {
     });
   }
 
+  async rollCreateCheckMessage(
+    flavor: string,
+    baseFormula: Formulator,
+    target: SerializedCompositeNumber,
+    options: RollOptions | undefined
+  ): Promise<void> {
+    const criticals = this.data.data.secondary.criticals;
+    const fullFormula = baseFormula.modify(options?.modifier).criticals({
+      success: criticals.success.total,
+      failure: criticals.failure.total
+    });
+    const checkRoll = new Roll(fullFormula.toString()).roll({ async: false });
+
+    const msgOptions = createDefaultMessageData(
+      ChatMessage.getSpeaker({ actor: this }),
+      options?.rollMode ?? getGame().settings.get("core", "rollMode")
+    );
+
+    msgOptions.flavor = flavor;
+
+    await diceSoNice(checkRoll, msgOptions.whisper, { actor: this.id });
+    const result = checkRoll.dice[0]?.results[0]?.result ?? 0;
+    const flags: CheckFlags = {
+      type: "roll",
+      details: {
+        criticals: {
+          success: criticals.success.toObject(false),
+          failure: criticals.failure.toObject(false)
+        },
+        success: target
+      },
+      roll: {
+        formula: checkRoll.formula,
+        critical: checkRoll.dice[0]?.results[0]?.critical,
+        result,
+        degree: fullFormula.d100Target - result,
+        total: checkRoll.total ?? 0
+      },
+      blind: msgOptions.blind ?? false
+    };
+    ChatMessage.create({
+      ...msgOptions,
+      flags: { [CONSTANTS.systemId]: flags }
+    });
+  }
+
   /**
    * Roll a SPECIAL for this Actor.
    * @param name - the name of the SPECIAL to roll
    */
   rollSpecial(name: SpecialName, options?: RollOptions): void {
-    const msgOptions: ConstructorDataType<foundry.data.ChatMessageData> = {
-      flavor: WvI18n.getSpecialRollFlavor(name),
-      speaker: ChatMessage.getSpeaker({ actor: this })
+    const targetRaw = this.data.data.specials[name];
+    const targetCompNum: SerializedCompositeNumber = {
+      source: targetRaw.permTotal,
+      components: targetRaw.tempComponents
```

Comment:
Wouldn't it be better to use the `source` and `[...targetRaw.permComponents, ...targetRaw.tempComponents]` here?

Reply:
Huh, I could have sworn `permComponents` was included in `tempComponents`.